### ISSUE TITLE
Type error: Argument 2 passed to Cloudflare\Api::request() must be of the type array, null given

### ISSUE THREAD

**Issue by aaronhuisinga** [jamesryanbell/cloudflare#73]
This is a pretty major bug introduced in version 1.11.0 which has caused us to need to roll back and lock our version at 1.10.3.

It appears that with the new parameter defaults for the request() method in v1.11.0, several calls are broken from methods that pass in null for the data parameter. We are getting this for every request made to the activation_check() method, for example.

```php
(new CloudFlareZone($this->cloudflare))->activation_check($external_id);
```

Obviously, you'd need to pass in values for the variables, but that is enough to throw an exception every time it is run. Several other methods are returning the same error. We're experiencing this while running PHP 7.1, though we have not tested with earlier versions.

I'd imagine that this bug is actually affecting all methods that don't pass in a data parameter to the request() method, so it seems pretty showstopping.

@jamesryanbell it seems like you'll need to simply update the default data parameter for the get(), patch(), put(), post(), delete()... etc methods in the Api class. It shouldn't be a difficult fix, I'd assume. Just an oversight when shipping the new version.

**Comment by aaronhuisinga** [comment]
@jamesryanbell I'll submit a PR that patches these methods, but I 100% recommend that you test several different portions of the API before deploying it, and maybe rolling back to v1.11.0 release. We only use a few methods in our application, so we won't be able to test everything.

**Comment by developer-devPHP** [comment]
@jamesryanbell  This is really a critical issue all requests not working in php7.1 

One more example
```php
(new Cloudflare\User\Billing\Subscriptions\Zones ())->list_zones()

 public function list_zones()
 {
        return $this->get('/user/billing/subscriptions/zones');
 }
```

Which is calling to API get 
```php
    public function get($path, array $data = null)
    {
        return $this->request($path, $data, 'get');
    }
```
Your request function is

```php
 protected function request($path, array $data = array(), $method = 'get')
 {
   // your code
 }

```

So this is global issue please fix it as soon as you can and before final push please test deeper, this is really big changes.

**Comment by aaronhuisinga** [comment]
@developer-devPHP I opened a PR to fix this in #74, though there has been no response.

@jamesryanbell please take a look at this ASAP, as v1.11.0 is completely broken for what will be a large percentage of users downloading it. The longer it takes to patch, the more users will update to the latest version and have a broken install.

**Comment by jamesryanbell** [comment]
Hi

I've deleted the 1.11.0 broken build and pulled @aaronhuisinga PR to fix the issue.

Thanks for the work everyone, apologises for the slow response.

Thanks,
James
